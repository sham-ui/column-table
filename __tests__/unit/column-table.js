import renderer from 'sham-ui-test-helpers';
import ColumnTable, { resolver } from '../../src/column-table.sfc';

it( 'renders correctly', () => {
    const meta = renderer( ColumnTable, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'resolver', () => {
    const meta = renderer( ColumnTable, {
        columns: [
            { title: 'Foo', data: 'foo' }
        ],
        items: [ 1, 2, 3 ],
        resolver: column => ( {
            ...resolver( column ),
            titleGetter: () => 'Bar'
        } )
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
