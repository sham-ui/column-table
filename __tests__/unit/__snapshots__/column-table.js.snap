// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "classNames": "",
    "columns": Array [],
    "items": Array [],
  },
  "html": "
<table class=\\"\\">
  <thead>
    <tr></tr>
  </thead>
  <tbody></tbody>
</table>
",
}
`;
